students/kristianjf/lesson09/mailroom_5.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+import sys
+'''
+Goal:
+You work in the mail room at a local charity. Part of your job is to write \
+incredibly boring, repetitive emails thanking your donors for their generous \
+gifts. You are tired of doing this over and over again, so you’ve decided to \
+let Python help you out of a jam and do your work for you.
+'''
+'''
+It should have a data structure that holds a list of your donors and a history of \
+the amounts they have donated. This structure should be populated at first with at \
+least five donors, with between 1 and 3 donations each.
+'''
+# self.DATA_STRUCTURE = [('Jeff Bezos', [200, 500, 1000]), \
+#                     ('Bill Gates', [1000]), \
+#                     ('Steve Jobs', [20, 50, 100])]
+# self.DATA_STRUCTURE_DICTS = [{'name': name, 'donation': donation} for name, donation in self.DATA_STRUCTURE]
+# # self.DATA_STRUCTURE_DICT = [ \
+# #                         {'name': 'Jeff Bezos', 'donation': [200, 500, 1000]},\
+# #                         {'name': 'Bill Gates', 'donation': [1000]},\
+# #                         {'name': 'Steve Jobs', 'donation': [20, 50, 100]}\
+# #                         ]
+
+class Donor:
+    def __init__(self, name, donations):
+        self._name = name
+        self._donations = [] if donations is None else donations
+    @property
+    def name(self):
+        return self._name
+    @property
+    def donations(self):
+        return self._donations
+    def donate(self, new_donation):
+        self.donations.append(new_donation)
+    def metrics(self):
```

Comment:
Probably better to have this computed and returned when you need it, rather than stored in self -- as these value will be out of date when a donation is added.

still good to do it all at once, and a method can return three values:

``return num_d total_d, avg_d`` 

alternatively, you could make them properties:

```
@property
def num_donations:
    return len(self._donations)
```